src/metabase/api/collection.clj

```diff
@@ -317,22 +326,26 @@
   and queries that are technically executable without parameters can be unacceptably slow
   without the necessary constraints. (Marking parameters in optional blocks as required doesn't
   seem to be useful any way, but if the user said it is required, we honor this flag.)"
-  [row]
+  [text template-tags]
+  (let [obligatory-params (into #{}
+                                (comp (filter params/Param?)
+                                      (map (comp keyword :k)))
+                                (params.parse/parse text))]
+    (and (every? #(or (:default %) (= (:type %) "dimension") (fully-parametrized-snippet? %))
```

Comment:
_If_ `:type` could be either a string or keyword wouldn't you need to take that into account here too?